src/cmd/main.go

```diff
@@ -205,6 +206,7 @@ func buildCheckMappings() {
 }
 
 func analyzeReports(diff []*linter.Report) (criticalReports int) {
+	filtered := diff[:0]
```

Comment:
To be honest, I am not a huge fan of micro-optimisations like this. analyzeReports probably is not a bottleneck so you are free to just allocate a new slice with enough capacity, for example. Code like this is harder to verify, at least for me, and it does not provide too much benefits in terms of performance in this case.